codecov_cli/services/commit/__init__.py

```diff
@@ -45,10 +45,8 @@ def send_commit_data(
 ):
     # this is how the CLI receives the username of the user to whom the fork belongs
     # to and the branch name from the action
-    tokenless = os.environ.get("TOKENLESS")
-    if tokenless:
+    if not token and branch and ":" in branch:
         headers = None  # type: ignore
-        branch = tokenless  # type: ignore
         logger.info("The PR is happening in a forked repo. Using tokenless upload.")
     else:
         headers = get_token_header_or_fail(token)
```

Comment:
strictly speaking, we no longer need any of this logic anymore. we just send whatever we have for the token and the branch, and the API will return a 401 if it's not allowed

the logs may still be helpful, so maybe change to:
```
if not token:
    if branch and ":" in branch:
        logger.info("Creating a commit on an unprotected branch")
    else:
        logger.warning("Token is missing, but branch is missing or protected")

headers = get_token_header(token)
```